### PR TITLE
Destination-MotherDuck: Use new table naming methodology, use explicit column names, and add debug info for column matching

### PR DESCRIPTION
**Pull request by aaronsteers**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from typing import Any, Dict, Iterable, List, Mapping
 from urllib.parse import urlparse
 
+from py import process
+
```
Comment (guenp) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py:16]:
```suggestion
```

Comment (guenp) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py:16]:
```suggestion
```

**Review by guenp — APPROVED** [review]

**Comment by aaronsteers** [comment]
Will hold this until we're ready to do the next release.

File: docs/integrations/destinations/motherduck.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ This connector is primarily designed to work with MotherDuck and local DuckDB fi
...
 |:--------| :--------- | :-------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| 0.1.4   | 2024-10-28 | [47688](https://github.com/airbytehq/airbyte/pull/47688)  | Use new destination table name format, and add debug info for column mismatches.                                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (guenp) [inline docs/integrations/destinations/motherduck.md:72]:
```suggestion
| 0.1.4   | 2024-10-28 | [47688](https://github.com/airbytehq/airbyte/pull/47688)  | Use new destination table name format, explicitly insert PyArrow table columns by name and add debug info for column mismatches.                                                                                                                                                                                                                                                                                                                                                                                 |
```

**Review by guenp — APPROVED** [review]

**Pull request merged**